packit_service/worker/allowlist.py

```diff
@@ -0,0 +1,346 @@
+
+    def issue_comment_event(
+        self,
+        event: Union[IssueCommentEvent, IssueCommentGitlabEvent],
+        project: GitProject,
+        service_config: ServiceConfig,
+        job_configs: Iterable[JobConfig],
+    ) -> bool:
+        account_name = event.user_login
+        if not account_name:
+            raise KeyError(f"Failed to get account_name from {type(event)}")
+        namespace = event.repo_namespace
+
+        namespace_approved = self.is_approved(namespace)
+        user_approved = project.can_merge_pr(account_name)
+
+        if namespace_approved and user_approved:
+            return True
+
+        msg = (
+            f"Namespace {namespace} is not on our allowlist!"
+            if not namespace_approved
+            else f"Account {account_name} has no write access!"
+        )
+        logger.error(msg)
+        project.issue_comment(event.issue_id, msg)
+        return False
+
+    def check_and_report(
+        self,
+        event: Optional[Any],
+        project: GitProject,
+        service_config: ServiceConfig,
+        job_configs: Iterable[JobConfig],
+    ) -> bool:
+        """
+        Check if account is approved and report status back in case of PR
+        :param service_config: service config
+        :param event: PullRequest and Release TODO: handle more
+        :param project: GitProject
+        :param job_configs: iterable of jobconfigs - so we know how to update status of the PR
+        :return:
+        """
+        CALLBACKS: Dict[
+            Union[type, Tuple[Union[type, Tuple[Any, ...]], ...]], Callable
+        ] = {
+            (  # events that are not checked against allowlist
+                PushPagureEvent,
+                PullRequestPagureEvent,
+                PullRequestCommentPagureEvent,
+                AbstractCoprBuildEvent,
+                TestingFarmResultsEvent,
+                DistGitEvent,
+                InstallationEvent,
+                KojiBuildEvent,
+            ): self.unchecked_event,
+            (
+                ReleaseEvent,
+                PushGitHubEvent,
+                PushGitlabEvent,
+            ): self.release_push_event,
+            (
+                PullRequestGithubEvent,
+                PullRequestCommentGithubEvent,
+                MergeRequestGitlabEvent,
+                MergeRequestCommentGitlabEvent,
+            ): self.pr_event,
+            (
+                IssueCommentEvent,
+                IssueCommentGitlabEvent,
+            ): self.issue_comment_event,
+        }
+
+        # Administrators
+        user_login = getattr(event, "user_login", None)
+        if user_login and user_login in service_config.admins:
+            logger.info(f"{user_login} is admin, you shall pass.")
+            return True
+
+        for related_events, callback in CALLBACKS.items():
+            if isinstance(event, related_events):
+                return callback(event, project, service_config, job_configs)
```

Comment:
Personally, I like that approach, but tbh, I am still not sure about the readability of this.

But let's use this and improve later if someone complains...;)